components/server/src/workspace/workspace-starter.ts

```diff
@@ -118,6 +119,63 @@ export interface StartWorkspaceOptions {
 const MAX_INSTANCE_START_RETRIES = 2;
 const INSTANCE_START_RETRY_INTERVAL_SECONDS = 2;
 
+export const migrationIDESettings = (user: User) => {
```

Comment:
@akosyakov This really feels we it's about time for an IDE service. :confused: 

I' rather not see this code under WebApp control, at all, but rather query your service for it. We could start out soft (with a top-level package in `server`, and a clean TS interface), but should not loose that out of sight.

I don't want to block this PR, but we should keep this in mind.

Reply:
https://github.com/gitpod-io/gitpod/issues/9189